src/hotspot/cpu/ppc/stubGenerator_ppc.cpp

```diff
@@ -631,6 +631,132 @@ class StubGenerator: public StubCodeGenerator {
     return start;
   }
 
+// Generate stub for ghash process blocks.
+//
+// Arguments for generated stub:
+//      state:  R3_ARG1
+//      subkeyH:    R4_ARG2
+//      data: R5_ARG3
+//      blocks: R6_ARG4
+//
+address generate_ghash_processBlocks() {
+  StubCodeMark mark(this, "StubRoutines", "ghash");
+  address start = __ function_entry();
+
+  // Registers for parameters
+  Register state = R3_ARG1;                     // long[] state
+  Register subkeyH = R4_ARG2;                   // long[] subH
+  Register data = R5_ARG3;                      // byte[] data
+  Register blocks = R6_ARG4;
+  Register temp1 = R8;
+  Register temp2 = R9;
+  Register temp3 = R10;
+  Register temp4 = R11;
+  Register align = data;
+  Register load = R12;
+  // Vector Registers
+  VectorRegister vZero = VR0;
+  VectorRegister vH = VR1;
+  VectorRegister vLowerH = VR2;
+  VectorRegister vHigherH = VR3;
+  VectorRegister vTmp4 = VR4;
+  VectorRegister vTmp5 = VR5;
+  VectorRegister vTmp6 = VR6;
+  VectorRegister vTmp7 = VR7;
+  VectorRegister vTmp8 = VR8;
+  VectorRegister vTmp9 = VR9;
+  VectorRegister vTmp10 = VR10;
+  VectorRegister vTmp11 = VR11;
+  VectorRegister vTmp12 = VR12;
+  VectorRegister loadOrder = VR13;
+  VectorRegister vHigh = VR14;
+  VectorRegister vLow = VR15;
+  VectorRegister vState = VR16;
+  VectorRegister vConstC2 = VR19;
+  Label L_end, L_aligned, L_error;
+
+  __ li(temp1, 0xc2);
+  __ sldi(temp1, temp1, 56);
+  __ vxor(vZero, vZero, vZero);
+  // Load the vector from memory into vConstC2
```

Comment:
This comment is not correct. It's not loaded from memory.

Reply:
Thanks for confirming that. So the low-order-bits of the reduction polynomial are in the high byte of `temp1` at this point, and the rest of `temp1` is zero, confirming that the polynomial (z^7+z^2+z+1) is in bit-reversed order:

```
+  __ li(temp1, 0xc2);
+  __ sldi(temp1, temp1, 56);
```